### ISSUE TITLE
User signup 

### ISSUE THREAD

**Issue by actuallyyun** [raul-sauco/hangboard-timer#3]
### user story
As a user, I want to sign up for the app. 

### Acceptance Criteria

- [ ] I should see a signup page with email, password, and username(?) field.
- [ ] I should see specific errors is my input does not pass the fe validation
- [ ] After successfully submit data, I should see a confirm email message

Question:
Should we provide signup with third-party account options? such as `Google`,`Facebook`,etc?

**Comment by raul-sauco** [comment]
> Question: Should we provide signup with third-party account options? such as `Google`,`Facebook`,etc?

Makes sense from the user's point of view, I just checked the Supabase website and they support social sign-in out of the box, let's, at least, look into it. If we decide later that is taking too much extra time we can scrape it and focus on other tasks.

**Comment by actuallyyun** [comment]
The supabase auth api was easy to use in my last project, we did not use social sign in options though
